microservices/optimizer/app.py

```diff
@@ -1,12 +1,20 @@
 """Optimizer: the optimization microservice. More information can be found
 in the documentation at `optimizer_api_doc.md`"""
 
-from flask import Flask, jsonify, Response, request
+from flask import Flask, jsonify, Response, request, g
 
-from utils import optimize
+from utils import DBManager, optimize
 
 app = Flask(__name__)
 
+DATABASE_FILE: str = "opt.db"
+
+
+@app.before_request
+def before_request() -> None:
+    """Adds the database manager to request context"""
+    g.db_manager = DBManager(DATABASE_FILE)
```

Comment:
This constructor call does not match the constructor in the `db_manager` module. The first argument should be the IP address and the second argument should be the table name. You should just replace it with a default constructor call:
```suggestion
    g.db_manager = DBManager()
```